elements/howto-accordion/howto-accordion.js

```diff
@@ -0,0 +1,591 @@
+    }
+
+    /**
+     * `disconnectedCallback` removes all the event listeners that
+     * `connectedCallback` added.
+     */
+    disconnectedCallback() {
+      this.removeEventListener('change', this._onChange);
+      this.removeEventListener('keydown', this._onKeyDown);
+    }
+
+    /**
+     * `_isHeading` returns true if the given element
+     * is a `<howto-accordion-heading>`.
+     */
+    _isHeading(elem) {
+      return elem.tagName.toLowerCase() === 'howto-accordion-heading';
+    }
+
+    /**
+     * `_onChange` handles the `change` event. The event’s
+     * target is the heading that has been instructed to expand by click,
+     * keyboard input.
+     */
+    _onChange(event) {
+      this._animatePanelForHeading(event.target, event.detail.isExpandedNow);
+    }
+
+    /**
+     * `_animatePanelForHeading` animates the expansion of a panel, provided
+     * there is no other animation running.
+     */
+    _animatePanelForHeading(heading, expand) {
+      // If there’s an animation running, ignore the event.
+      if (this.classList.contains('animating'))
+        return;
+      const panel = this._panelForHeading(heading);
+      if(expand) {
+        this._expandPanel(panel);
+        this._animateIn(panel);
+      } else {
+        this._animateOut(panel)
+          .then(_ => this._collapsePanel(panel));
+      }
+    }
+
+    /**
+     * `_onKeyDown` handles key presses inside the accordion.
+     */
+    _onKeyDown(event) {
+      // If the currently focused element is not a heading, the keypress
+      // originated from inside a panel or empty space. Nothing to do.
+      const currentHeading = event.target;
+      if (!this._isHeading(currentHeading)) return;
+      // Don’t handle modifier shortcuts typically used by assistive technology.
+      if (event.altKey) return;
+
+      // The switch-case will determine which heading should be focused next
+      // depending on the key that was pressed.
+      let newHeading;
+      switch (event.keyCode) {
+        case KEYCODE.LEFT:
+        case KEYCODE.UP:
+          newHeading = this._prevHeading();
+          break;
+
+        case KEYCODE.RIGHT:
+        case KEYCODE.DOWN:
+          newHeading = this._nextHeading();
+          break;
+
+        case KEYCODE.HOME:
+          newHeading = this._firstHeading();
+          break;
+
+        case KEYCODE.END:
+          newHeading = this._lastHeading();
+          break;
+        // Any other key press is ignored and passed back to the browser.
+        default:
+          return;
+      }
+
+      // The browser might have some native functionality bound to the arrow
+      // keys, home or end. The element calls `preventDefault` to prevent the
+      // browser from taking any actions.
+      event.preventDefault();
+      // Make the currently focused heading unfocusable, then make the new
+      // heading focusable and give focus to it.
+      currentHeading._shadowButton.setAttribute('tabindex', -1);
```

Comment:
Add/remove `tabindex=-1` from the headings themselves rather than interacting with their shadow dom internals